src/tokenizedSubaccounts/PPTSA.sol

```diff
@@ -0,0 +1,578 @@
+    PPTSAStorage storage $ = _getPPTSAStorage();
+    int markCostOfTrade = lowerStrike.markPrice.multiplyDecimal(lowerStrike.tradeAmount)
+      + higherStrike.markPrice.multiplyDecimal(higherStrike.tradeAmount);
+
+    int actualCostOfTrade = lowerStrike.tradePrice.toInt256().multiplyDecimal(lowerStrike.tradeAmount)
+      + higherStrike.tradePrice.toInt256().multiplyDecimal(higherStrike.tradeAmount);
+
+    /*
+     * We want to ensure that if we are trading spreads, the cost we're trading isn't too far off from the mark cost.
+     * For long spreads, we don't want to pay too much over the mark cost,
+     * so maxTotalCostTolerance is a number over 1 (1e18),
+     * and if our projected cost is over that tolerance we revert since the trade is too expensive to be reasonable.
+     *
+     * For short spreads, its the reverse. We don't want to receive too little money for the spread.
+     * The same logic applies, but the maxTotalCostTolerance is a number under 1 (1e18).
+     * if our projected cost is under that tolerance we revert since the trade is too cheap to be reasonable.
+     */
+    if (
+      $.isLongSpread
+        && actualCostOfTrade.abs() > markCostOfTrade.abs().multiplyDecimal($.ppParams.maxTotalCostTolerance)
+    ) {
+      revert PPT_TotalCostOverTolerance();
+    } else if (
+      !$.isLongSpread
+        && actualCostOfTrade.abs() < markCostOfTrade.abs().multiplyDecimal($.ppParams.maxTotalCostTolerance)
+    ) {
+      revert PPT_TotalCostBelowTolerance();
+    }
+
+    uint markValueToStrikeDiffRatio =
+      (lowerStrike.markPrice - higherStrike.markPrice).abs().divideDecimal(higherStrike.strike - lowerStrike.strike);
+
+    /*
+     * This check ensures that the (markPrice difference) / (strike Difference) ratio is within a certain range.
+     * Typically, the range will be between ~.4 and ~.6, but this can be adjusted in the params.
+     * If the ratio is too high then we are buying a spread that is too expensive in respect to the strikes.
+     * If too low, the trade is not as risky. This is to force the vault to only make realistic slightly-risky trades.
+     */
+    if (
+      markValueToStrikeDiffRatio < $.ppParams.minMarkValueToStrikeDiffRatio
+        || markValueToStrikeDiffRatio > $.ppParams.maxMarkValueToStrikeDiffRatio
+    ) {
+      revert PPT_MarkValueNotWithinBounds();
+    }
+  }
+
+  function _getOptionPrice(uint optionExpiry, uint optionStrike)
+    internal
+    view
+    returns (uint callPrice, uint putPrice, uint callDelta)
+  {
+    uint timeToExpiry = optionExpiry - block.timestamp;
+    (uint vol, uint forwardPrice) = _getFeedValues(optionStrike.toUint128(), optionExpiry.toUint64());
+
+    return Black76.pricesAndDelta(
+      Black76.Black76Inputs({
+        timeToExpirySec: timeToExpiry.toUint64(),
+        volatility: vol.toUint128(),
+        fwdPrice: forwardPrice.toUint128(),
+        strikePrice: optionStrike.toUint128(),
+        discount: 1e18
```

Comment:
whats this discount variable? Are we sure it doesn't flip depending on `isLongSpread`, `isCall`, `isMaker` ?

Reply:
This is a value to do with black76 computation, should be fixed at 1 (hardcoded that way in the managers themselves too)